airflow/utils/log/file_task_handler.py

```diff
@@ -0,0 +1,171 @@
+        :param try_number: task instance try_number to read logs from. If None
+                           it returns all logs separated by try_number
+        :return: a list of logs
+        """
+        current_try = task_instance.try_number
+
+        if try_number is None:
+            try_numbers = list(range(current_try))
+        elif try_number > current_try or try_number < 0:
+            logs = ['Error fetching the logs. Try number {} is invalid.'.format(try_number)]
+            return logs
+        else:
+            try_numbers = [try_number]
+
+        logs = [''] * len(try_numbers)
+        for i, try_number in enumerate(try_numbers):
+            logs[i] += self._read(task_instance, try_number)
+
+        return logs
+
+    def _init_file(self, ti):
+        """
+        Create log directory and give it correct permissions.
+        :param ti: task instance object
+        :return relative log path of the given task instance
+        """
+        # To handle log writing when tasks are impersonated, the log files need to
+        # be writable by the user that runs the Airflow command and the user
+        # that is impersonated. This is mainly to handle corner cases with the
+        # SubDagOperator. When the SubDagOperator is run, all of the operators
+        # run under the impersonated user and create appropriate log files
+        # as the impersonated user. However, if the user manually runs tasks
+        # of the SubDagOperator through the UI, then the log files are created
+        # by the user that runs the Airflow command. For example, the Airflow
+        # run command may be run by the `airflow_sudoable` user, but the Airflow
+        # tasks may be run by the `airflow` user. If the log files are not
+        # writable by both users, then it's possible that re-running a task
+        # via the UI (or vice versa) results in a permission error as the task
+        # tries to write to a log file created by the other user.
+        relative_path = self.filename_template.format(
+            dag_id=ti.dag_id, task_id=ti.task_id,
+            execution_date=ti.execution_date.isoformat(), try_number=ti.try_number + 1)
+        full_path = os.path.join(self.local_base, relative_path)
+        directory = os.path.dirname(full_path)
+        # Create the log file and give it group writable permissions
+        # TODO(aoen): Make log dirs and logs globally readable for now since the SubDag
+        # operator is not compatible with impersonation (e.g. if a Celery executor is used
+        # for a SubDag operator and the SubDag operator has a different owner than the
+        # parent DAG)
+        if not os.path.exists(directory):
+            # Create the directory as globally writable using custom mkdirs
+            # as os.makedirs doesn't set mode properly.
+            mkdirs(directory, 0o775)
+
+        if not os.path.exists(full_path):
+            open(full_path, "a").close()
+            os.chmod(full_path, 0o666)
```

Comment:
@aoen ok fully understood and I agree. Besides I think it won't be required to do this anyway, if we let tasks log to stdout and let the log file be handled by the LocalTaskJob as I mentioned earlier.